eng/packages/http-client-csharp-mgmt/emitter/src/resource-detection.ts

```diff
@@ -150,18 +153,20 @@ function parseResourceOperation(
 
 function getParentResourceModelId(
   sdkContext: SdkContext<AzureEmitterOptions, SdkHttpOperation>,
-  model: SdkModelType | undefined
+  model: SdkModelType | undefined,
+  program: Program
 ): string | undefined {
   const decorators = (model?.__raw as Model)?.decorators;
   const parentResourceDecorator = decorators?.find(
     (d) => d.definition?.name == parentResourceName
   );
-  return getResourceModelId(sdkContext, parentResourceDecorator) ?? undefined;
+  return getResourceModelId(sdkContext, parentResourceDecorator, program) ?? undefined;
 }
 
 function getResourceModelId(
   sdkContext: SdkContext<AzureEmitterOptions, SdkHttpOperation>,
-  decorator?: DecoratorApplication
+  decorator?: DecoratorApplication,
+  program?: Program
```

Comment:
I think `program` here will never be undefined from the caller side. We could just make this parameter required and remove the null check below